client/platform/desktop/backend/serializers/viame.ts

```diff
@@ -0,0 +1,273 @@
+/**
+ * VIAME CSV parser/serializer copied logically from
+ * viame_server.serializers.viame python module
+ */
+
+import csvparser from 'csv-parse';
+import fs from 'fs-extra';
+import { pipeline } from 'stream';
+
+import {
+  TrackData, Feature, StringKeyObject, ConfidencePair, TrackSupportedFeature,
+} from 'vue-media-annotator/track';
+
+const CommentRegex = /^\s*#/g;
+const HeadRegex = /^\(kp\) head ([0-9]+\.*[0-9]*) ([0-9]+\.*[0-9]*)/g;
+const TailRegex = /^\(kp\) tail ([0-9]+\.*[0-9]*) ([0-9]+\.*[0-9]*)/g;
+const AttrRegex = /^\(atr\) (.*?)\s(.+)/g;
+const TrackAttrRegex = /^\(trk-atr\) (.*?)\s(.+)/g;
+const PolyRegex = /^(\(poly\)) ((?:[0-9]+\.*[0-9]*\s*)+)/g;
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll
+function getCaptureGroups(regexp: RegExp, str: string) {
+  const array = [...str.matchAll(regexp)];
+  if (array.length) {
+    return array[0];
+  }
+  return null;
+}
+
+function _rowInfo(row: string[]) {
+  if (row.length < 9) {
+    throw new Error('malformed row: too few columns');
+  }
+  if (row[0].match(CommentRegex) !== null) {
+    throw new Error('malformed row: comment row');
+  }
+  return {
+    trackId: parseInt(row[0], 10),
+    filename: row[1],
+    frame: parseInt(row[2], 10),
+    bounds: (
+      row.slice(3, 7).map((v) => Math.round(parseFloat(v))) as
+      [number, number, number, number]
+    ),
+    fishLength: parseFloat(row[8]),
+  };
+}
+
+function _deduceType(value: string): boolean | number | string {
+  if (value === 'true') {
+    return true;
+  }
+  if (value === 'false') {
+    return false;
+  }
+  try {
+    return parseFloat(value);
+  } catch {
+    return value;
+  }
+}
```

Comment:
This may need to be modified because It will return `NaN` for some string values if it can't properly parse the float.

Reply:
I forgot how the parseFloat api worked.  It doesn't throw an error, it just returns NaN.